include/seqan3/alignment/pairwise/policy/unbanded_score_trace_dp_matrix_policy.hpp

```diff
@@ -0,0 +1,114 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::unbanded_score_trace_dp_matrix.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <range/v3/view/zip.hpp>
+
+#include <seqan3/alignment/matrix/trace_directions.hpp>
+#include <seqan3/alignment/pairwise/policy/unbanded_dp_matrix_policy.hpp>
+#include <seqan3/std/span.hpp>
+
+namespace seqan3::detail
+{
+/*!\brief Manages the allocation and provision of an unbanded dynamic programming matrix.
+ * \ingroup alignment_policy
+ * \tparam derived_t   The derived alignment algorithm.
+ * \tparam score_allocator_t The allocator type used for allocating the score matrix.
+ * \tparam trace_allocator_t The allocator type used for allocating the trace matrix.
+ */
+template <typename derived_t, typename score_allocator_t, typename trace_allocator_t>
+class unbanded_score_trace_dp_matrix_policy :
+    public unbanded_dp_matrix_policy<unbanded_score_trace_dp_matrix_policy<derived_t,
+                                                                           score_allocator_t,
+                                                                           trace_allocator_t>,
+                                     score_allocator_t>
+{
+private:
+
+    //!\brief The base type
+    using base_t = unbanded_dp_matrix_policy<unbanded_score_trace_dp_matrix_policy<derived_t,
+                                                                                   score_allocator_t,
+                                                                                   trace_allocator_t>,
+                                             score_allocator_t>;
+
+    //!\brief Befriends the derived class to grant it access to the private members.
+    friend derived_t;
+    //!\brief Make dimension_first_range visible in this class.
+    using base_t::dimension_first_range;
```

Comment:
why not `horizontal` and `vertical` range?

Reply:
I tried to keep the naming with respect to the user interface, which uses first_range and second_range. Only in some specific places where I refer to the matrix layout I have a comment talking about horizontal and vertical direction. I see I could do it for the dimensions as well, but mostly when using the dimensions the ranges are affected by this.